pandas/core/arrays/base.py

```diff
@@ -236,6 +239,7 @@ class ExtensionArray:
     # ------------------------------------------------------------------------
 
     @classmethod
+    @abstractmethod
```

Comment:
This might be a fairly large change since EA authors will now need to actually make sure their unused methods are at least defined?

Reply:
Can make them non-abstract and raise `NotImplementedError`. Are there more classes that should rather not be abstract?

Reply:
> Can make them non-abstract and raise NotImplementedError

Yeah I think this would be better

> Are there more classes that should rather not be abstract?

IMO anything that is in the public API would be better to raise a `NotImplementedError`

Reply:
ExtensionArray/Dtype are no longer abstract.